### ISSUE TITLE
vis_graph() and vis_graph_sf() is returning a list of 1 and a dataframe of 0 obs. of 4 variables.

### ISSUE THREAD

**Issue by magnusol9** [jmlondon/pathroutr#17]
I am trying to apply this workflow to a different marine mammal, but I am hitting a snag. 

Firstly, I get a warning which I do not know if is related.

> land_region <- sf::st_buffer(dugongsARGOS_sf, dist = 500000) %>% 
+   sf::st_union() %>% 
+   sf::st_convex_hull() %>% 
+   sf::st_intersection(newcaledoniacoast) %>% 
+   st_collection_extract('POLYGON') %>% 
+   st_sf()
Warning message:
In st_collection_extract.sfc(., "POLYGON") : x is already of type POLYGON.

The next plot functions fine. However, when I run

vis_graph <- land_region %>%
  prt_visgraph()

I get a list of 1

And when I run I get a data frame of 0 obs. of 4 variables 
vis_graph_sf <- sfnetworks::activate(vis_graph,"edges") %>% 
  sf::st_as_sf() 

These values do not allow themselves to the next plot. Is there an obvious snag I am missing?

**Comment by jmlondon** [comment]
@magnusol9 

I suspect the issue is that the resulting sf object from:

```r
land_region <- sf::st_buffer(dugongsARGOS_sf, dist = 500000) %>%
sf::st_union() %>%
sf::st_convex_hull() %>%
sf::st_intersection(newcaledoniacoast)
```
is already a 'POLYGON' and you do not need the `st_collection_extract()` step. 

What happens if you skip that step and go straight to `st_sf()` ??

**Comment by magnusol9** [comment]
Thank you for your reply. I tried without st_sf(), but the same issue. My shapefile, '1' and 'land_region' are all also '1 obs. of 1 variable' if that may be what is causing the issue.

**Comment by jmlondon** [comment]
@magnusol9, thank you for providing your data to me directly.

I was actually puzzled for a bit until I realized that your coordinate values and the specified projection mismatched. While the objects (`newcaledoniacoast` and `dugongsARGOS_sf`) state they are in the _RGNC91-93 / Lambert New Caledonia_ projection, the actual coordinates in the geometry are still geographic (long/lat). `pathroutr` requires projected coordinates -- I do some low-level testing for this but I didn't anticipate a mismatch. So, `prt_visgraph()` returns an empty visibility network when passed geographic coordinates.

My suspicious is that somewhere along the way `sf::st_set_crs()` was used to specify (overwrite) the projection instead of `sf::st_transform()` which will _transform_ from one projection to the other. 

Below, I've provided some code with your data to show how I was able to return a non-empty visgraph

```r
library(pathroutr)
library(sf)
load('~/Downloads/newcaledoniacoast.rda')
load('~/Downloads/dugongsARGOS_sf.rda')

newcaledoniacoast <- newcaledoniacoast %>% 
  st_set_crs(4326) %>% 
  st_transform(3163)
dugongsARGOS_sf <- dugongsARGOS_sf %>% 
  st_set_crs(4326) %>% 
  st_transform(3163)

land_region <- sf::st_buffer(dugongsARGOS_sf, dist = 500000) %>%
  sf::st_union() %>%
  sf::st_convex_hull() %>%
  sf::st_intersection(newcaledoniacoast) %>%
  st_sf()

vis_graph <- land_region %>%
  prt_visgraph()
```
